framework/validators/UniqueValidator.php

```diff
@@ -133,7 +140,14 @@ public function validateAttribute($model, $attribute)
                 $this->addError($model, $attribute, Yii::t('yii', '{attribute} is invalid.'));
                 return;
             }
-            $conditions[] = [$key => $value];
+            if ($this->ignoreCase)
+            {
+                $newKey         = "lower(" . $key . ")";
+                $newValue       = "lower('" . $value . "')";
```

Comment:
allowing to inject, see more https://www.yiiframework.com/doc/api/2.0/yii-db-expression

Reply:
> allowing to inject, see more https://www.yiiframework.com/doc/api/2.0/yii-db-expression

I've only been a Yii user for a couple of weeks now, and didn't know this yet - I'm still going through the documentation. This, though, **is incredibly useful**. Thank's for the suggestion and the code example.

Reply:
This seems like a good idea, but I'm having a problem understanding it. The easiest way to get my head around it, is to look at what it's doing in terms of the final sql passed to the database, so I'll do that as soon as I can.

Do you have any advice for **where** the expression should be created? Right there within the ```if ($this->ignoreCase)``` block as I have it, or elsewhere. 

I also want to change the way the query is done at the moment. It currently produces this code:
```SELECT EXISTS(SELECT * FROM "User" WHERE lower("User"."UserEmail")=lower('NewUser@Example.com'))```

It would be better if it produced this code:
```SELECT EXISTS(SELECT lower("User"."UserEmail") FROM "User" WHERE lower("User"."UserEmail")=lower('NewUser@Example.com'))```

While it does not look like much of a change, on a larger dataset, it would cause the planner to choose to do an index scan rather than a table scan, though that would change from DB to DB.